test/tck/steps/bolt_type_steps.py

```diff
@@ -0,0 +1,140 @@
+        context.L.append(tck_util.get_bolt_value(row[0], row[1]))
+
+
+@step("adding a table of values to the map M")
+def step_impl(context):
+    for row in context.table:
+        context.M['a' + str(len(context.M))] = tck_util.get_bolt_value(row[0], row[1])
+
+
+@step("adding map M to list L")
+def step_impl(context):
+    context.L.append(context.M)
+
+
+@when("adding a table of lists to the map M")
+def step_impl(context):
+    for row in context.table:
+        context.M['a' + str(len(context.M))] = tck_util.get_list_from_feature_file(row[1], row[0])
+
+
+@step("adding a copy of map M to map M")
+def step_impl(context):
+    context.M['a' + str(len(context.M))] = context.M.copy()
+
+
+@when("the driver asks the server to echo this value back")
+def step_impl(context):
+    context.results = {}
+    context.results["as_string"] = tck_util.send_string("RETURN " + tck_util.as_cypger_text(context.expected))
+    context.results["as_parameters"] = tck_util.send_parameters("RETURN {input}", {'input': context.expected})
+
+
+@when("the driver asks the server to echo this list back")
+def step_impl(context):
+    context.expected = context.L
+    context.results = {}
+    context.results["as_string"] = tck_util.send_string("RETURN " + tck_util.as_cypger_text(context.expected))
+    context.results["as_parameters"] = tck_util.send_parameters("RETURN {input}", {'input': context.expected})
+
+
+@when("the driver asks the server to echo this map back")
+def step_impl(context):
+    context.expected = context.M
+    context.results = {}
+    context.results["as_string"] = tck_util.send_string("RETURN " + tck_util.as_cypger_text(context.expected))
+    context.results["as_parameters"] = tck_util.send_parameters("RETURN {input}", {'input': context.expected})
+
+
+@then("the result returned from the server should be a single record with a single value")
+def step_impl(context):
+    assert len(context.results) > 0
+    for result in context.results.values():
+        assert len(result) == 1
+        assert len(result[0]) == 1
+
+
+@step("the value given in the result should be the same as what was sent")
+def step_impl(context):
+    assert len(context.results) > 0
+    for result in context.results.values():
+        result_value = result[0].values()[0]
+        assert result_value == context.expected
+
+
+@step("the node value given in the result should be the same as what was sent")
+def step_impl(context):
+    assert len(context.results) > 0
+    for result in context.results.values():
+        result_value = result[0].values()[0]
+        assert result_value == context.expected
+        assert result_value.labels == context.expected.labels
+        assert result_value.keys() == context.expected.keys()
+        assert result_value.values() == context.expected.values()
+        assert result_value.items() == context.expected.items()
+        assert len(result_value) == len(context.expected)
+        assert iter(result_value) == iter(context.expected)
```

Comment:
Is `iter(...) == iter(...)` actually what you want to do here? the LHS and RHS will both return generators and this will compare the generator objects themselves rather than the contents.

You probably want to change `iter` to `list` which will consume the iterables and compare as lists instead.